src/benchmarks/micro/libraries/Common/StreamBuffer.cs

```diff
@@ -338,7 +338,9 @@ private void CancelWaiter()
                 if (Interlocked.Exchange(ref _hasWaiter, 0) == 1)
                 {
                     Debug.Assert(_waitSourceCancellationToken != default);
-		    _waitSource.SetException(ExceptionDispatchInfo.SetCurrentStackTrace(new OperationCanceledException(_waitSourceCancellationToken)));
+#if NET5_0_OR_GREATER // API introduced in .NET 5
```

Comment:
without it, I was not able to compile the project for `netcoreapp3.1`:

```log
PS D:\projects\performance\src\benchmarks\micro> dotnet build -c Release -f netcoreapp3.1 .\MicroBenchmarks.csproj
Microsoft (R) Build Engine version 17.0.0-preview-21378-03+d592862ed for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  Reporting -> D:\projects\performance\artifacts\bin\Reporting\Release\netstandard2.0\Reporting.dll
  BenchmarkDotNet.Extensions -> D:\projects\performance\artifacts\bin\BenchmarkDotNet.Extensions\Release\netstandard2.0\BenchmarkDotNet.Extensions.dll
D:\projects\performance\src\benchmarks\micro\libraries\Common\StreamBuffer.cs(341,54): error CS0117: 'ExceptionDispatchInfo' does not contain a definition for 'SetCurrentStackTrace' [D:\projects\performance\src\benchmarks\micro\MicroBenchmarks.csproj]

Build FAILED.

D:\projects\performance\src\benchmarks\micro\libraries\Common\StreamBuffer.cs(341,54): error CS0117: 'ExceptionDispatchInfo' does not contain a definition for 'SetCurrentStackTrace' [D:\projects\performance\src\benchmarks\micro\MicroBenchmarks.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.88
```

this API was introduced in .NET 5: https://apisof.net/catalog/e04e14a6-c36f-2f98-8763-37884ea5b9da but for some reason our CI has missed this problem (most probably we don't have 3.1 CI leg anymore, which imo is not a big deal)